tests/runner/test_sequential_runner.py

```diff
@@ -278,3 +279,77 @@ def test_confirms(self, mocker, pipeline, is_async):
         catalog = DataCatalog(data_sets={"ds1": fake_dataset_instance})
         SequentialRunner(is_async=is_async).run(pipeline, catalog)
         fake_dataset_instance.confirm.assert_called_once_with()
+
+
+@pytest.fixture
+def resume_scenario_pipeline():
+    return Pipeline(
+        [
+            node(identity, "ds0_A", "ds1_A", name="node1_A"),
+            node(identity, "ds0_B", "ds1_B", name="node1_B"),
+            node(
+                multi_input_list_output,
+                ["ds1_A", "ds1_B"],
+                ["ds2_A", "ds2_B"],
+                name="node2",
+            ),
+            node(identity, "ds2_A", "ds3_A", name="node3_A"),
+            node(identity, "ds2_B", "ds3_B", name="node3_B"),
+            node(identity, "ds3_A", "ds4_A", name="node4_A"),
+            node(identity, "ds3_B", "ds4_B", name="node4_B"),
+        ]
+    )
+
+
+@pytest.fixture
+def resume_scenario_catalog():
+    def _load():
+        return 0
+
+    def _save(arg):
+        assert arg == 0
+
+    persistent_dataset = LambdaDataSet(load=_load, save=_save)
+    return DataCatalog(
+        {
+            "ds0_A": persistent_dataset,
+            "ds0_B": persistent_dataset,
+            "ds2_A": persistent_dataset,
+            "ds2_B": persistent_dataset,
+        }
+    )
+
+
+@pytest.mark.parametrize(
+    "failing_node_indexes,expected_pattern",
+    [
+        ([0], r"No nodes ran."),
```

Comment:
Would it be possible to label these by node name rather than index? e.g. `["node1_A"]` here and then maybe `failing_nodes = pipeline.only_nodes(*failing_node_names)`. This way you don't have to understand how node index maps onto the pipeline topology.

Reply:
Very good point, I have implemented this in [1bf5f1d](https://github.com/kedro-org/kedro/pull/1795/commits/1bf5f1de48ff9634b2c207212db0afb23f537b06).